src/Constraint/PropertiesConstraint.php

```diff
@@ -63,8 +63,37 @@ public function normalize(stdClass $schema, Context $context, Walker $walker)
      */
     public function apply($instance, stdClass $schema, Context $context, Walker $walker)
     {
-        $this->validateInstance($instance, $schema, $context);
-        $this->validateChildren($instance, $schema, $context, $walker);
+        // implementation of the algorithms described in 5.4.4.4 and in 8.3
+        foreach ($instance as $property => $value) {
+            $context->enterNode($instance->{$property}, $property);
+            $schemas = [];
+
+            if (isset($schema->properties->{$property})) {
+                $schemas[] = $schema->properties->{$property};
+            }
+
+            foreach ($schema->patternProperties as $regex => $propertySchema) {
+                if (Utils::matchesRegex($property, $regex)) {
+                    $schemas[] = $propertySchema;
+                }
+            }
+
+            if (!$schemas) {
```

Comment:
Scrutinizer is righty complaining about this. Please use `empty` or `count` instead of an implicit conversion.